apps/console/src/features/applications/components/edit-application.tsx

```diff
@@ -176,18 +180,51 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
      * Called when the defaultActiveIndex updates.
      */
     useEffect( () => {
-        // Change defaultActiveIndex value for custom applications.
-        if ((application?.templateId === CustomApplicationTemplate.id
-            || application?.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_OIDC
-            || application?.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_PASSIVE_STS
-            || application?.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_SAML)
-            && !urlSearchParams.get(ApplicationManagementConstants.APP_STATE_PROTOCOL_PARAM_KEY)) {
-            setActiveTabIndex(defaultActiveIndex - 1);
+        if(isEmpty(window.location.hash) ){
+            handleDefaultTabIndex();
+        }
+        /**
+         * Keep track of authentication callback URLs to handle ordinary application edit section view use case and the
+         * application edit section view after refreshing the page separately.
+         *
+         */
+        const authenticationCallbackUrl = AuthenticateUtils.getAuthenticationCallbackUrl(AppConstantsCore.CONSOLE_APP);
+
+        callbackUrlList.push(authenticationCallbackUrl);
+        setCallbackUrlList(callbackUrlList);
+
+        if(callbackUrlList.length===2 && !callbackUrlList[0].includes(`#${URLFragmentTypes.TAB_INDEX}`)) {
+            handleDefaultTabIndex();
+        }
+    },[defaultActiveIndex]);
+
+    /**
+     * Called when the URL fragment updates
+     */
+    useEffect( () => {
+        if(totalTabs === undefined || window.location.hash.includes(URLFragmentTypes.VIEW) ||
+            isEmpty(window.location.hash)) {
             return;
         }
-        setActiveTabIndex(defaultActiveIndex);
 
-    },[defaultActiveIndex]);
+        const urlFragment:string[] = window.location.hash.split("#"+URLFragmentTypes.TAB_INDEX);
+
+        if(urlFragment.length === 2 && isEmpty(urlFragment[0]) && /^\d+$/.test(urlFragment[1])) {
+            const tab_index:number = parseInt(urlFragment[1]);
```

Comment:
Missing `radix` param.
```suggestion
            const tab_index:number = parseInt(urlFragment[1]);
```